services/app-api/handlers/reports/release.test.ts

```diff
@@ -0,0 +1,116 @@
+import { APIGatewayProxyEvent } from "aws-lambda";
+import { proxyEvent } from "../../utils/testing/proxyEvent";
+import { StatusCodes } from "../../utils/types/types";
+import {
+  mockDocumentClient,
+  mockDynamoDataMLRLocked,
+  mockDynamoDataMLRComplete,
+} from "../../utils/testing/setupJest";
+import { releaseReport } from "./release";
+import KSUID from "ksuid";
+import { error } from "../../utils/constants/constants";
+
+jest.mock("../../utils/auth/authorization", () => ({
+  isAuthorized: jest.fn().mockResolvedValue(true),
+  hasPermissions: jest.fn(() => {}),
+}));
+
+const mockAuthUtil = require("../../utils/auth/authorization");
+
+jest.mock("../../utils/debugging/debug-lib", () => ({
+  init: jest.fn(),
+  flush: jest.fn(),
+}));
+
+const mockProxyEvent: APIGatewayProxyEvent = {
+  ...proxyEvent,
+  headers: { "cognito-identity-id": "test" },
+  pathParameters: { reportType: "MLR", state: "AB", id: "testReportId" },
+  body: JSON.stringify(mockDynamoDataMLRComplete),
+};
+
+const releaseEvent: APIGatewayProxyEvent = {
+  ...mockProxyEvent,
+};
+
+describe("Test releaseReport method", () => {
+  beforeEach(() => {
+    // fail state and pass admin auth checks
+    mockAuthUtil.hasPermissions
+      .mockReturnValueOnce(true)
+      .mockReturnValueOnce(true)
+      .mockReturnValueOnce(true)
```

Comment:
Can we find another way to do this? There is no context on when to return true vs false and is dependent on 5 tests running in the correct order.

Reply:
Agreed. I am new to jest mocking (previously would just do full e2e tests or write methods around stuff I couldn't run in test) so I was just following conventions here. Any thoughts on how this could be improved?

Reply:
I'll take a look at get back to you. I don't know for certain either :D

Reply:
the majority of the time we want permissions to return true, so setting that as the default response when we create mockAuthUtil

`const mockAuthUtil = require("../../utils/auth/authorization");
mockAuthUtil.hasPermissions.mockReturnValue(true);`

then later when we need it to return false we override it to be false
`test("Test release report without admin permissions throws 403", async () => {
    mockAuthUtil.hasPermissions.mockReturnValueOnce(false);
    mockDocumentClient.get.promise.mockReturnValueOnce({
      Item: mockDynamoDataMLRLocked,
    });
    const res = await releaseReport(releaseEvent, null);
    expect(res.statusCode).toBe(StatusCodes.UNAUTHORIZED);
    expect(res.body).toContain(error.UNAUTHORIZED);
  });`

This allows us to delete the `beforeEach` function call from the `describe` block

Reply:
@tblackwe Makes sense! I'm going to write another PR towards `mlr` that has those changes, since this PR is closed already.